### PR TITLE
[nodejs] respect lackof TTYs

### PR DESCRIPTION
**Pull request by AllAwesome497**
For anyone who's unfamiliar with the term TTY, its essentially what makes a terminal a terminal (as opposed to a file that's just read/written from/to).  Generally speaking, we want TTYs since they're what handle colors and terminal escapes.  However for machine-read input/output, we don't want TTYs as we don't want the program we're running to send us escapes (as a result of believing its talking to a TTY).

Ran into two issues with nodejs w/ IO testing.

- We print a bunch of extra things (eg, a "hint" to hit ctrl+c to exit) that we don't want w/ io tests.
- `confirm` and `prompt` (functions we define) use `readline-sync` which doesn't support running without a TTY, meaning neither would work with IO testing.

### CODE REVIEW

File: prybar_assets/nodejs/input-sync.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+  const previousRawMode = process.stdin.isRaw;
+
+  try {
+    process.stdin.setRawMode(true);
+
+    return cb();
+  } finally {
+    process.stdin.setRawMode(previousRawMode);
+  }
```
Comment (masad-frost) [inline prybar_assets/nodejs/input-sync.js:67]: Finally always messes with my head when it comes to flow control, I find it hard to reason about.

Wonder if it's easier to read if this function returned a callback to reset to previous raw mode.

```js
function question(prompt) {
  const resetRawMode = ensureRawMode();
  ...promptcode...
  resetRawMode();
  return str;
}
```

**Review by masad-frost — APPROVED** [review]
Sweet, thanks for comments and tests.

File: prybar_assets/nodejs/input-sync.js (hunk 1 above)
Comment (AllAwesome497) [reply prybar_assets/nodejs/input-sync.js:67]: I added the finally + made it like this for 2 reasons:
1. We should reset it even if something throws (but throw the error anyways)
2. Its easy to miss restoring the tty's state.  This is the closest JS equivalent to golang's defer that I can think of.

Comment (AllAwesome497) [reply prybar_assets/nodejs/input-sync.js:67]: I'll move the return outside of the try/finally block, since I can see how that gets confusing.

Comment (masad-frost) [reply prybar_assets/nodejs/input-sync.js:68]: Yeah that's better, thanks

**Pull request merged**